llvm/lib/Transforms/Scalar/GVN.cpp

```diff
@@ -3329,7 +3343,8 @@ class llvm::gvn::GVNLegacyPass : public FunctionPass {
     AU.addPreserved<TargetLibraryInfoWrapperPass>();
     AU.addPreserved<LoopInfoWrapperPass>();
     AU.addRequired<OptimizationRemarkEmitterWrapperPass>();
-    AU.addPreserved<MemorySSAWrapperPass>();
+    if (Impl.isMemorySSAEnabled())
+      AU.addRequired<MemorySSAWrapperPass>();
```

Comment:
(legacy pass so doesn't matter much, but..) MSSA is always preserved with the above change (if getAnalysisIfAvailable is kept)

Reply:
I meant that this line: `AU.addPreserved<MemorySSAWrapperPass>();` should be kept here, not removed.
But the diff to add the conditional required is good, could you readd please?
```
if (Impl.isMemorySSAEnabled())
      AU.addRequired<MemorySSAWrapperPass>();
```

Reply:
Oops, had accidentally dropped it, restored it now, thanks for the catch!